### PR TITLE
Code linting

### PR DESCRIPTION
**Pull request by gnott**
Along the lines of the linting I did in the ``jats-generator`` project, the major changes are reducing the number of object properties and refactoring functions out of the class that have no self-use.

Checked and with a high score as reported by ``proofreader``.

This PR is welcome for review or comment - no obligation to look too closely. The test fixtures and scenarios are almost completely unaltered and the tests pass.

### CODE REVIEW

File: tests/test_generate.py [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
 from elifepubmed.conf import config, parse_raw_config
-from elifearticle.utils import unicode_value
+
 
 TEST_BASE_PATH = os.path.dirname(os.path.abspath(__file__)) + os.sep
 TEST_DATA_PATH = TEST_BASE_PATH + "test_data" + os.sep
 generate.TMP_DIR = TEST_BASE_PATH + "tmp" + os.sep
 
+
+def read_file_content(file_name):
```
Comment (seanwiseman) [inline tests/test_generate.py:14]: Can just be:
```
def read_file_content(file_name):
    with open(file_name, 'rb') as file_p:
        return file_p.read()
```
The `with` context manager will handle the tidy up of the file.

File: tests/test_generate.py [hunk 2]
```diff
@@ -36,20 +45,14 @@ def setUp(self):
             ('pb369-jats.xml', 'pb-pubmed-369-20170717071707.xml',
              'pb', self.default_pub_date))
 
-    def read_file_content(self, file_name):
```
Comment (seanwiseman) [inline tests/test_generate.py:39]: The same could have been achieved using:
```
@staticmethod
def read_file_content(file_name):
    ...
```
If you still wanted to use it as a class method, but either is fine. One benefit is if you were to inherit from `TestGenerate` elsewhere (I don't know if you do) you would have `self.read_file_content()` available without having to import it again. 

I normally handle all generic file parsing in my test fixtures and pass in the actual parsed values to the test cases, it isolates the logic a bit more and reduces unnecessary multiple parses of the same file if the contents don't change.

This is just opinion, there's nothing technically wrong with your current approach.

File: elifepubmed/utils.py [hunk 3]
```diff
@@ -15,10 +16,12 @@ def allowed_tags():
         '<p>', '</p>'
     )
 
+
 def allowed_tag_names():
```
Comment (seanwiseman) [inline elifepubmed/utils.py:20]: Could be:
```
import re
...

def allowed_tag_names():
    return list(sorted(set([re.sub('[<>/]', '', tag) for tag in allowed_tags()])))
```

File: tests/test_generate.py (hunk 1 above)
Comment (gnott) [reply tests/test_generate.py:14]: Agreed, I am converting these when I see them - thanks!

File: tests/test_generate.py (hunk 2 above)
Comment (gnott) [reply tests/test_generate.py:39]: I see, I didn't completely understand ``@staticmethod`` and its uses. Tests in other libraries I have found these fixture reading functions are helpful to live outside of the test classes and they can be imported anyway, I think one place I put these helper functions right in ``__init__.py`` in the tests folder. I can follow what ever people prefer if there becomes a preference.

File: elifepubmed/utils.py (hunk 3 above)
Comment (gnott) [reply elifepubmed/utils.py:20]: Yes, I like it. This is only a first pass at linting, and there is always room for improvement!

**Review by seanwiseman — APPROVED** [review]

**Pull request merged**